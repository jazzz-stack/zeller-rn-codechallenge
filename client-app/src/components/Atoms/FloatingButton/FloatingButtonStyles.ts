import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  keyboardAvoidingContainer: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    pointerEvents: 'box-none', // Allow touches to pass through except for the button
  },
  button: {
    width: 55,
    height: 55,
    justifyContent: 'center',
    alignItems: 'center',
    elevation: 8,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    borderRadius: 50,
  },

  // Variants
  primary: {
    backgroundColor: '#107fdaff',
  },
  secondary: {
    backgroundColor: '#F2F2F7',
    borderWidth: 1,
    borderColor: '#C6C6C8',
  },
  danger: {
    backgroundColor: '#FF3B30',
  },

  // Sizes
  small: {
    paddingHorizontal: 12,
    paddingVertical: 8,
    minHeight: 32,
  },
  medium: {
    paddingHorizontal: 16,
    paddingVertical: 12,
    minHeight: 44,
  },
  large: {
    paddingHorizontal: 20,
    paddingVertical: 16,
    minHeight: 52,
  },

  // Text styles
  buttonText: {
    fontSize: 16,
    fontWeight: '600',
    textAlign: 'center',
  },
  primaryText: {
    color: '#FFFFFF',
  },
  secondaryText: {
    color: '#007AFF',
  },
  dangerText: {
    color: '#FFFFFF',
  },
});
