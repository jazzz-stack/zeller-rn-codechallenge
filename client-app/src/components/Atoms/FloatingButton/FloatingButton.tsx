import React from 'react';
import {TouchableOpacity, TouchableOpacityProps, View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { styles } from './FloatingButtonStyles';

interface FloatingButtonProps extends TouchableOpacityProps {
  variant?: 'primary' | 'secondary' | 'danger';
  size?: 'small' | 'medium' | 'large';
  iconName?: string;
  iconSize?: number;
  iconColor?: string;
}

const FloatingButton: React.FC<FloatingButtonProps> = ({
  variant = 'primary',
  size = 'medium',
  iconName = 'person-add',
  iconSize = 25,
  iconColor = '#ffffff',
  style,
  ...props
}) => {
  return (
    <View
      style={styles.keyboardAvoidingContainer}
      pointerEvents="box-none">
      <TouchableOpacity
        style={[styles.button, styles[variant], styles[size], style]}
        {...props}>
        <Icon name={iconName} size={iconSize} color={iconColor} />
      </TouchableOpacity>
    </View>
  );
};

export default FloatingButton;
