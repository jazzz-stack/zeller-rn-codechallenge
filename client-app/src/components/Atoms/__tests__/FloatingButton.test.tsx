import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import FloatingButton from '../FloatingButton/FloatingButton';

// Mock react-native-vector-icons
jest.mock('react-native-vector-icons/MaterialIcons', () => 'MaterialIcons');

describe('FloatingButton', () => {
  it('should render with testID', () => {
    const {getByTestId} = render(<FloatingButton testID="floating-button" />);
    const button = getByTestId('floating-button');

    expect(button).toBeTruthy();
  });

  it('should pass testID prop correctly', () => {
    const {getByTestId} = render(<FloatingButton testID="my-custom-test-id" />);
    const button = getByTestId('my-custom-test-id');

    expect(button).toBeTruthy();
  });

  it('should handle onPress with testID', () => {
    const onPressMock = jest.fn();
    const {getByTestId} = render(
      <FloatingButton testID="floating-button" onPress={onPressMock} />
    );
    const button = getByTestId('floating-button');

    fireEvent.press(button);

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('should render with all custom props', () => {
    const {getByTestId} = render(
      <FloatingButton 
        testID="floating-button"
        variant="danger"
        size="large"
        iconName="add"
        iconSize={30}
        iconColor="#000000"
      />
    );
    const button = getByTestId('floating-button');

    expect(button).toBeTruthy();
  });

  // Snapshot Tests
  it('matches snapshot with default props', () => {
    const component = render(<FloatingButton testID="floating-button" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('matches snapshot with onPress handler', () => {
    const onPressMock = jest.fn();
    const component = render(
      <FloatingButton testID="floating-button" onPress={onPressMock} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
