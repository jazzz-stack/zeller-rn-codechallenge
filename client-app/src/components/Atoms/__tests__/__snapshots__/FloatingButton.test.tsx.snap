// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatingButton matches snapshot with default props 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "bottom": 20,
      "pointerEvents": "box-none",
      "position": "absolute",
      "right": 20,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#107fdaff",
        "borderRadius": 50,
        "elevation": 8,
        "height": 55,
        "justifyContent": "center",
        "minHeight": 44,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
        "width": 55,
      }
    }
    testID="floating-button"
  >
    <MaterialIcons
      color="#ffffff"
      name="person-add"
      size={25}
    />
  </View>
</View>
`;

exports[`FloatingButton matches snapshot with onPress handler 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "bottom": 20,
      "pointerEvents": "box-none",
      "position": "absolute",
      "right": 20,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#107fdaff",
        "borderRadius": 50,
        "elevation": 8,
        "height": 55,
        "justifyContent": "center",
        "minHeight": 44,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4.65,
        "width": 55,
      }
    }
    testID="floating-button"
  >
    <MaterialIcons
      color="#ffffff"
      name="person-add"
      size={25}
    />
  </View>
</View>
`;
